bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -178,9 +261,38 @@ public void propertyChange(PropertyChangeEvent evt) {
 			}
 
 			// Sky brightness warnings
+
 			Double sb = a.getMin(Circumstance.TOTAL_SKY_BRIGHTNESS, false);
-			if (sb != null && !a.getObs().getConditions().containsSkyBrightness(sb))
+
+			final Option<Interval> skyBackgroundInterval = night.flatMap(n -> {
+				final Obs obs = a.getObs();
+				final SkyBackgroundSolver skyBackgroundSolver = new SkyBackgroundSolver(site, obs::getCoords, sb);
+				final SortedSet<Interval> intervals = skyBackgroundSolver.solve(n).getIntervals();
+				if (intervals.isEmpty()) {
+					return None.instance();
+				} else {
+					final Interval interval = intervals.first();
+					if (interval.getStart() >= n.getStart() && interval.getEnd() <= n.getEnd()) {
+						return None.instance();
+					}
+					return new Some<>(intervals.first());
```

Comment:
Is the idea that the `skyBackgroundInterval` be empty if the solver finds an interval that falls inside of the night? Could you add a comment or something that explains what this block does and what it will be used for?

Reply:
That was a last moment hack. For some reason the default sky background interval is larger than the night interval (concretely the end is different, the start is exactly the same).